src/workerd/api/actor-destroy.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/debug.h>
+
+#include <workerd/api/basics.h>
+#include <workerd/io/worker-interface.capnp.h>
+#include <workerd/io/worker-interface.h>
+
+namespace workerd::api {
+
+class ActorDestroyEvent final : public ExtendableEvent {
+public:
+  explicit ActorDestroyEvent();
+
+  static jsg::Ref<ActorDestroyEvent> constructor(kj::String type) = delete;
+
+  JSG_RESOURCE_TYPE(ActorDestroyEvent) {
+    JSG_INHERIT(ExtendableEvent);
+  }
+};
+
+class ActorDestroyCustomEventImpl final : public WorkerInterface::CustomEvent,
+                                          public kj::Refcounted {
+public:
+  ActorDestroyCustomEventImpl(uint16_t typeId) : typeId(typeId) {}
+
+  kj::Promise<Result> run(kj::Own<IoContext_IncomingRequest> incomingRequest,
+                          kj::Maybe<kj::StringPtr> entrypointName) override;
+
+  kj::Promise<Result> sendRpc(capnp::HttpOverCapnpFactory& httpOverCapnpFactory,
+                              capnp::ByteStreamFactory& byteStreamFactory,
+                              kj::TaskSet& waitUntilTasks,
+                              rpc::EventDispatcher::Client dispatcher) override;
+
+  uint16_t getType() override {
+    return typeId;
+  }
+
+private:
+  uint16_t typeId;
+};
+
+#define EW_ACTOR_DESTROY_ISOLATE_TYPES api::ActorDestroyEvent, api::ActorDestroyExportedHandler
```

Comment:
It seems like these are not actually used...  Might be left over from earlier development, or specific to service-worker-style workers, that might (intentionally?) not be supported for destroy?

If it is important for V8 to know about the registered types, the macro needs to included in the combined isolate declaration (e.g. try grepping for `EW_QUEUE_ISOLATE_TYPES`).